### PR TITLE
Vendor in Libnetwork changes

### PR DESCRIPTION
**Pull request by arkodg**
Bring in changes from https://github.com/moby/libnetwork/pull/2572 to moby

Signed-off-by: Arko Dasgupta <arko.dasgupta@docker.com>

### CODE REVIEW

**Comment by arkodg** [comment]
cc @bboehmke

**Comment by thaJeztah** [comment]
Seeing a failure that may be relevant;

```
 WARN [runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/go/src/github.com/docker/docker/integration-cli/docker_cli_daemon_test.go:948:15: Exists not declared by package iptables /go/src/github.com/docker/docker/integration-cli/docker_cli_daemon_test.go:948:70: Exists not declared by package iptables /go/src/github.com/docker/docker/integration-cli/docker_cli_daemon_test.go:953:14: Exists not declared by package iptables /go/src/github.com/docker/docker/integration-cli/docker_cli_daemon_test.go:953:68: Exists not declared by package iptables] 
```

File: vendor/github.com/docker/libnetwork/iptables/iptables.go [hunk 1]
```diff
@@ -380,35 +426,36 @@ func (c *ChainInfo) Output(action Action, args ...string) error {
...
-	Raw("-t", string(c.Table), "-X", c.Name)
+	iptable.Raw("-t", string(c.Table), "-F", c.Name)
+	iptable.Raw("-t", string(c.Table), "-X", c.Name)
 	return nil
 }
 
 // Exists checks if a rule exists
-func Exists(table Table, chain string, rule ...string) bool {
-	return exists(false, table, chain, rule...)
+func (iptable IPTable) Exists(table Table, chain string, rule ...string) bool {
```
Comment (thaJeztah) [inline vendor/github.com/docker/libnetwork/iptables/iptables.go:445]: Looks like this is causing the failure

Comment (bboehmke) [reply vendor/github.com/docker/libnetwork/iptables/iptables.go:445]: To reduce code duplication the code of the iptables was reused for ip6tables.

A simple change in the `integration-cli/docker_cli_daemon_test.go` should fix this:
```diff
diff --git a/integration-cli/docker_cli_daemon_test.go b/integration-cli/docker_cli_daemon_test.go
index d5fa41c29e..95a396cb1b 100644
--- a/integration-cli/docker_cli_daemon_test.go
+++ b/integration-cli/docker_cli_daemon_test.go
@@ -945,12 +945,14 @@ func (s *DockerDaemonSuite) TestDaemonLinksIpTablesRulesWhenLinkAndUnlink(c *tes
 
        sourceRule := []string{"-i", bridgeName, "-o", bridgeName, "-p", "tcp", "-s", childIP, "--sport", "80", "-d", parentIP, "-j", "ACCEPT"}
        destinationRule := []string{"-i", bridgeName, "-o", bridgeName, "-p", "tcp", "-s", parentIP, "--dport", "80", "-d", childIP, "-j", "ACCEPT"}
-       if !iptables.Exists("filter", "DOCKER", sourceRule...) || !iptables.Exists("filter", "DOCKER", destinationRule...) {
+
+       iptable := iptables.GetIptable(iptables.IPv4)
+       if !iptable.Exists("filter", "DOCKER", sourceRule...) || !iptables.Exists("filter", "DOCKER", destinationRule...) {
                c.Fatal("Iptables rules not found")
        }
 
        s.d.Cmd("rm", "--link", "parent/http")
-       if iptables.Exists("filter", "DOCKER", sourceRule...) || iptables.Exists("filter", "DOCKER", destinationRule...) {
+       if iptable.Exists("filter", "DOCKER", sourceRule...) || iptables.Exists("filter", "DOCKER", destinationRule...) {
                c.Fatal("Iptables rules should be removed when unlink")
        }
 

```

**Review by AkihiroSuda — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**